frameworks/PHP/reactphp/app.php

```diff
@@ -1,134 +1,149 @@
 <?php
+
+use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\ServerRequestInterface as Request;
+use React\EventLoop\Loop;
+use React\MySQL\ConnectionInterface as DbConnection;
+use React\MySQL\Factory as DbFactory;
 use React\Http\Message\Response;
+use React\MySQL\QueryResult;
+use React\Promise\PromiseInterface;
 
-function init()
-{
-    global $world, $fortune, $update;
-    $pdo = new PDO(
-        'mysql:host=tfb-database;dbname=hello_world',
-        'benchmarkdbuser',
-        'benchmarkdbpass',
-        [
-            PDO::ATTR_DEFAULT_FETCH_MODE => PDO::FETCH_ASSOC,
-            PDO::ATTR_EMULATE_PREPARES   => false
-        ]
-    );
-    $world   = $pdo->prepare('SELECT id,randomNumber FROM World WHERE id=?');
-    $update  = $pdo->prepare('UPDATE World SET randomNumber=? WHERE id=?');
-    $fortune = $pdo->prepare('SELECT id,message FROM Fortune');
-    $fortune->setFetchMode(PDO::FETCH_KEY_PAIR);
-}
+use function React\Promise\all;
 
-function router(Request $request)
+/** @return Closure(Request):ResponseInterface */
+function requestHandler(): Closure
 {
-    return match($request->getUri()->getPath()) {
-        '/plaintext' => text(),
-        '/json'      => json(), 
-        '/db'        => db(),
-        '/fortunes'  => fortune(),
-        '/query'     => query($request),
-        '/update'    => updateraw($request),
-        // '/info'      => info(),
-        default      => new Response(404, [], 'Error 404'),
+    $connection = establishDbConnection('benchmarkdbuser:benchmarkdbpass@tfb-database/hello_world?idle=0.5');
+
+    $world = static function (int $id) use ($connection): PromiseInterface {
+        return $connection->query('SELECT id,randomNumber FROM World WHERE id=?', [$id]);
     };
-}
 
-function text()
-{
-    return new Response(200, [
-        'Content-Type' => 'text/plain'
-    ], 'Hello, World!');
-}
+    $fortune = static function () use ($connection): PromiseInterface {
+        return $connection->query('SELECT id,message FROM Fortune');
+    };
```

Comment:
Seems like a waste creating these every request, maybe a `global` or a singleton *shivers* is in order here?

Reply:
This isn't being created on each request, though.

The function scope and returned closure behaves as a singleton.

Reply:
:facepalm: Just noticed it's returning a closure, you're right about here and `$connection`. Ignore this you're right :+1: . Going to feel stupid for a couple of minutes :joy: .

Reply:
HAHAHAHA I imagined that the closure was indeed being missed from your review 😂

There's no need for feeling stupid, I can't think of anyone who never overlooked things ❤️

Reply:
For good results, here we need to only execute the prepared statement.

Reply:
> For good results, here we need to only execute the prepared statement.

Even better. 0.7 will come with a connection pool: https://github.com/friends-of-reactphp/mysql/issues/175